src/main/java/dp/xlsx/DatasetFormatter.java

```diff
@@ -160,16 +167,32 @@ private void addHeaderRow(Collection<String> timeLabels) {
 
     private void addTimeLabelCells(Collection<String> timeLabels, Row headerRow, int columnOffset) {
 
+        final String[] additionalHeaders = file.getAdditionalHeaders();
+
         // write time labels across the title row
         for (String timeLabel : timeLabels) {
             Cell cell = headerRow.createCell(columnOffset);
-            cell.setCellStyle(valueRightAlign);
+            cell.setCellStyle(workBookStyles.getValueRightAlignStyle());
             cell.setCellValue(timeLabel);
             columnOffset++;
+
+            for (String additionalHeader : additionalHeaders) {
+                addHeaderCell(columnOffset, headerRow, additionalHeader + " (" + timeLabel + ")");
```

Comment:
I can only imagine that this cell will be the widest in the column. If so we will have to make sure the column width is as wide as the label

Reply:
I raised the column factor size